src/gleam/regex.gleam

```diff
@@ -0,0 +1,122 @@
+//// This module contains regular expression matching functions for strings.
+//// The matching algorithms of the library are based on the PCRE library, but not
+//// all of the PCRE library is interfaced and some parts of the library go beyond
+//// what PCRE offers. Currently PCRE version 8.40 (release date 2017-01-11) is used.
+
+import gleam/list
+import gleam/option.{Option}
+
+pub external type Regex
+
+/// The details about a particular match:
+///
+/// - match — the full string of the match.
+/// - index — the index of the match in the original string.
+/// - number — each match is numbered, starting from 1
+/// - submatches — a Regex can have subpatterns, sup-parts that are in parentheses.
+///
+pub type Match {
+  Match(
+    match: String,
+    index: Int,
+    number: Int,
+    submatches: List(Option(String)),
+  )
+  Error(Nil)
+}
+
+/// Create a new Regex.
+///
+/// ## Examples
+///
+///    > let re = from_string("[0-9]")
+///    >   |> result.unwrap(never())
+///    > match(re, "abc123")
+///    True
+///
+///    > match(re, "abcxyz")
+///    False
+///
+pub external fn from_string(String) -> Result(Regex, Nil) =
+  "gleam_stdlib" "regex_from_string"
+
+/// A regular expression that never matches any string.
+pub external fn never() -> Regex =
+  "gleam_stdlib" "regex_never"
```

Comment:
What is this function for? I've not come across one like this before.

Reply:
https://package.elm-lang.org/packages/elm/regex/latest/Regex#never , but I don't think we need it.